test/box-luatest/downgrade_test.lua

```diff
@@ -973,3 +973,40 @@ g.test_downgrade_global_names = function(cg)
         t.assert_equals(#box.space._cluster:format(), 2)
     end)
 end
+
+-----------------------------
+-- Check downgrade from 3.1.0
+-----------------------------
+
+g.test_downgrade_func_trigger = function(cg)
+    cg.server:exec(function()
+        local helper = require('test.box-luatest.downgrade_helper')
+        local trigger = require('trigger')
+        local _func = box.space._func
+        local prev_version = helper.prev_version(helper.app_version('3.0.0'))
+        local lua_code = 'function(a, b) return a + b end'
+        t.assert_equals(trigger.info('test'), {})
+        box.schema.func.create('trigger_func',
+            {body = lua_code, trigger = 'test'})
+        local errmsg = 'Function trigger_func is registered as ' ..
+            'event trigger. It is supported starting from version 3.1.0'
+        t.assert_equals(box.schema.downgrade_issues(prev_version), {errmsg})
+        -- Check if cannot actually downgrade
+        t.assert_error_msg_content_equals(errmsg, box.schema.downgrade,
+            prev_version)
```

Comment:
Let's at least test for `errmsg`. `prev_version` can be from issue of a different kind.

Also let's test that we don't drop 20-th column by accident (as under certain conditions we drop it).

I would also check that `non_trigger_function` is not updated in this case if present.

Reply:
Done.